lua/nvim_lsp/cssls.lua

```diff
@@ -0,0 +1,56 @@
+local skeleton = require 'nvim_lsp/skeleton'
+local util = require 'nvim_lsp/util'
+local lsp = vim.lsp
+
+local server_name = "cssls"
+local bin_name = "css-languageserver"
+
+local installer = util.npm_installer {
+  server_name = server_name;
+  packages = { "vscode-css-languageserver-bin" };
+  binaries = {bin_name};
+}
+
+skeleton[server_name] = {
+  default_config = util.utf8_config {
+    cmd = {bin_name, "--stdio"};
+    filetypes = {"css", "scss", "less"};
+    root_dir = util.root_pattern("package.json");
```

Comment:
I think perhaps  `root_dir = vim.loop.os_homedir;` makes more sense here in case a user is writing CSS outside of a node project, which is not uncommon.

Reply:
You could do an or condition. rootpattern or os_homedir

Reply:
something like...

```lua
root_dir = util.path.exists(util.root_pattern("package.json")) ? util.root_pattern("package.json") : vim.loop.os_homedir;
```


Not too familiar with lua, so not sure if that is correct of not.

Reply:
In lua terms, you can save the root pattern in a variable above and capture it in a function for root_dir

so

```lua

local root_pattern = util.root_pattern("package.json")

skeleton.cssls = {
  root_dir = function(fname)
    return root_pattern(fname) or vim.loop.os_homedir()
  end
}
```

Reply:
Today I learned...

Will update.